ttt.c

```diff
@@ -176,6 +179,17 @@ char check_win(char *t)
     return 'D';
 }
 
+int cmp_moves(const void *a, const void *b)
+{
+    int score_a = 0, score_b = 0;
+
+    if (history_count[*(int *) a])
```

Comment:
Declare two variables whose type is `int *`, so that you can avoid casting here. That is,
```c
int *_a = (int *) a, *_b = (int *) b;
```